logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.log4j.spi.RootLogger;
+
+import com.google.common.base.Function;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.lang.NumberFormatException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Properties;
+
+/**
+ * A centralized log server for Alluxio
+ *
+ * Alluxio masters and workers generate logs and store the logs in local storage. {@link AlluxioLogServerProcess}
+ * allows masters and workers to "push" their logs to a centralized log server where another copy of the logs will
+ * be stored.
+ */
+public class AlluxioLogServerProcess implements LogServerProcess {
+  private ServerSocket mServerSocket;
+  private int mPort;
+  private String mBaseLogDir;
+  private HashMap<InetAddress, Hierarchy> mInetAddressHashMap;
+  private boolean mStopped;
+
+  public AlluxioLogServerProcess(String portStr, String baseLogDir) {
+    try {
+      mPort = Integer.parseInt(portStr);
+      mBaseLogDir = baseLogDir;
+      mServerSocket = new ServerSocket(mPort);
+      mInetAddressHashMap = new HashMap<>();
+      mStopped = false;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+  /**
+   * Starts the Alluxio process. This call blocks until the process is stopped via
+   * {@link #stop()}. The {@link #waitForReady()} method can be used to make sure that the
+   * process is ready to serve requests.
+   */
+  @Override
+  public void start() throws Exception {
+    while (!mStopped) {
+      Socket client = mServerSocket.accept();
+      InetAddress inetAddress = client.getInetAddress();
+      // Currently loggerRepository is always null because we are not using the map.
+      LoggerRepository loggerRepository = mInetAddressHashMap.get(inetAddress);
+      if (loggerRepository == null) {
+        loggerRepository = configureHierarchy(inetAddress);
+      }
+      try {
+        new Thread(new AlluxioLog4jSocketNode(client, loggerRepository)).start();
+      } catch (IOException e) {
+        // Could not create a thread to serve a client, ignore this client.
+        continue;
+      }
+    }
+  }
+
+  /**
+   * Stops the Alluxio process, blocking until the action is completed.
+   */
+  @Override
+  public void stop() throws Exception {
+    mStopped = true;
+  }
+
+  /**
+   * Waits until the process is ready to serve requests.
+   */
+  @Override
+  public void waitForReady() {
+    CommonUtils.waitFor(this + " to start", new Function<Void, Boolean>() {
+      @Override
+      public Boolean apply(Void input) {
+        return mServerSocket != null;
+      }
+    }, WaitForOptions.defaults().setTimeoutMs(10000));
+  }
+
+  private LoggerRepository configureHierarchy(InetAddress inetAddress) throws IOException, URISyntaxException {
+    Hierarchy clientHierarchy;
+    String inetAddressStr = inetAddress.toString();
```

Comment:
instead of parsing the output of `toString`, could we use `getHostAddress` instead?

Reply:
Done.